core/src/main/scala/stryker4s/run/EtaCalculator.scala

```diff
@@ -0,0 +1,43 @@
+package stryker4s.run
+
+import grizzled.slf4j.Logging
+
+import scala.concurrent.duration._
+import language.postfixOps
+
+class EtaCalculator(amountOfMutants: Int) extends Logging {
+  val runResults: Array[Long] = new Array(amountOfMutants)
+  private[this] val stream = Iterator.range(0, amountOfMutants)
+
+  def time[T](fun: => T, log: String => Unit): T = {
+    val startTime = System.currentTimeMillis()
+
+    // Run the function
+    val result: T = fun
+
+    // Measure duration and store in array
+    val duration = System.currentTimeMillis() - startTime
+    val runNr = stream.next()
+    runResults(runNr) = duration
+
+    // Log estimate
+    log(calculateETA(runNr + 1))
+
+    result
+  }
+
+  def calculateETA(currentRun: Int): String = {
+    val from = math.max(0, currentRun - 5) // Base estimate on last 5 runs
+    val average = runResults.slice(from, currentRun).sum / (currentRun - from)
```

Comment:
Why the average from the last 5 runs? If a couple of the last runs time out this will really skew the results. The median of all runs makes more sense to me.

Reply:
The idea with taking the last 5 runs is basically we say: "If things keep going like this, it will take X minutes". Taking the median of all runs could result in skewed results as well. 

For example: What if halfway through a run you boot up a different program that eats up most of your RAM and CPU? Stryker will slow down significantly, but the estimate will take a long time to adjust because you've already done so many runs, so the average will hardly change. Resulting in a estimate consistently lower than the actual time. 

Maybe 5 runs is not enough, maybe it should be 10? Still I would argue this is better than taking the median.

We would have to think of (or find) some formula that takes all runs in consideration but does not count the first few runs with equal 'severity' as the more recent runs if we'd really want to do this properly. 🤔

Reply:
Hmm, good point. I do feel like 5 is too little. 10 is probably good. I do still feel like the median of the last 10 would be better than the average. To filter out any timeouts/outliers.

Reply:
I changed it to 10 and used median instead of average. 👍